src/controllers/hid/hidioreport.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "controllers/controller.h"
+#include "controllers/hid/hiddevice.h"
+#include "util/duration.h"
+
+class HidIoReport {
+  public:
+    HidIoReport(const unsigned char& reportId,
+            hid_device* device,
+            std::shared_ptr<const mixxx::hid::DeviceInfo> deviceInfo);
+    void sendOutputReport(QByteArray data);
+
+  private:
+    const unsigned char m_reportId;
+    const RuntimeLoggingCategory m_logOutput;
+    hid_device* const
```

Comment:
How do you guarantee that an instance of `HidIoReport` does not outlive the structure referenced by `m_pHidDevice`? The `hid_device` struct is created by the C API while `HidIoReport` is created by application code. Their lifetimes are decoupled.

In my opinion `HidIoReport` should only carry the report ID and the payload data. If you need to access context information those functions should to be placed in a context that has access to both, the report as well as the device information.

Both `hid_device*` and `const mixxx::hid::DeviceInfo&` could be passed as parameters to `sendOutputReport()` by the caller, i.e. the thread. There is no need to store references to this shared context data (immutable members of the thread class) redundantly in each and every report afaik.

Reply:
Thanks for the explaining words, they helped me to understand your requests and I will address it!
But currently I'm working on addressing Daniels remarks above, this will be commited first.